### PR TITLE
Update PHP_CodeSniffer

### PR DESCRIPTION
**Pull request by dplewis**
I've added a few commands to the package.json file.

`npm test` - shortcut for tests
`npm run test:coverage` - for travis
`npm run lint` - Lints src and test folders
`npm run lint:fix` - Automatically Lints files

I've also forced travis to lint files.

`CONTIBUTING.md` has been updated

I have phpcs already installed on my text editor so I thought it would be nice to have in this project.

### CODE REVIEW

File: src/Parse/ParseClient.php [hunk 1]
```diff
@@ -124,6 +124,7 @@
      *
      * @throws Exception
      */
+    //@codingStandardsIgnoreLine
```
Comment (montymxb) [inline src/Parse/ParseClient.php]: Just curious what was the error phpcs was bringing up here?

Comment (montymxb) [inline src/Parse/ParseClient.php]: Nvm I took a look at it myself. Considering that this line is rather long I would suggest breaking it down like `ParseClient::_request`. Then just make sure to remove the ignore above it.

**Comment by dplewis** [comment]
@montymxb for future reference if you want to ignore lines for linter

```
// @codingStandardsIgnoreStart
echo 'ignore';
// @codingStandardsIgnoreEnd
```

```
// @codingStandardsIgnoreLine
echo 'ignore line' // @codingStandardsIgnoreLine
```

If you want to ignore lines to improve coverage

```
// @codeCoverageIgnoreStart
echo 'coverage';
// @codeCoverageIgnoreEnd
```

```
exit; // @codeCoverageIgnore
```

File: src/Parse/ParseObject.php [hunk 2]
```diff
@@ -807,6 +806,7 @@ public static function destroyAll(array $objects, $useMasterKey = false)
      */
     private static function destroyBatch(array $objects, $useMasterKey = false)
     {
+        //@codingStandardsIgnoreStart
```
Comment (montymxb) [inline src/Parse/ParseObject.php]: Same thing here, you can break down the offending string being set on `path`. Looking it you can probably break it down fairly easily like so. Just be sure to remove the ignore start/end once done.
```php
$data[] = [
	'method' => 'DELETE',
	'path'   =>
		'/'.ParseClient::getMountPath().
		'classes/'.$object->getClassName().
		'/'.$object->getObjectId(),
];
```

File: src/Parse/ParseUser.php [hunk 3]
```diff
@@ -192,9 +192,10 @@ public static function logInWithTwitter(
         $screen_name,
         $consumer_key,
         $consumer_secret = null,
-        $auth_token,
-        $auth_token_secret)
-    {
+        $auth_token, //@codingStandardsIgnoreLine
```
Comment (montymxb) [inline src/Parse/ParseUser.php]: This is fine for an ignore, but if you can put it between like so
```php
$consumer_secret = null,
//@codingStandardsIgnoreLine
$auth_token,
```
it should still work and won't be trailing after `$auth_token` instead.

File: src/Parse/ParseUser.php [hunk 4]
```diff
@@ -246,10 +247,11 @@ public static function loginWithAnonymous()
          * @link https://en.wikipedia.org/wiki/Universally_unique_identifier
          */
         $uuid_parts = str_split(md5(mt_rand()), 4);
+        //@codingStandardsIgnoreStart
```
Comment (montymxb) [inline src/Parse/ParseUser.php]: Same, please break the offending line down and remove the ignores.

File: src/Parse/ParseUser.php [hunk 5]
```diff
@@ -338,10 +342,11 @@ public function linkWithTwitter(
         $screen_name,
         $consumer_key,
         $consumer_secret = null,
-        $auth_token,
+        $auth_token, //@codingStandardsIgnoreLine
```
Comment (montymxb) [inline src/Parse/ParseUser.php]: Same as the other twitter one, just bump it between the two rather than right after one of the args.

**Comment by montymxb** [comment]
I don't think we'll ever be needing to ignore lines to bump up our coverage, besides I wouldn't really have any desire in artificially inflating that number. 

According to [codecov's report](https://codecov.io/gh/parse-community) we're already the top dog at just under 98% total coverage as is 👍 .

**Comment by flovilmart** [comment]
Yeah, don't ignore the not-covered lines it makes it impossible to improve correctly the coverage or identifying potential issues.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -63,7 +63,15 @@ Alternately you can configure a compatible test server as follows:
...
-    ./vendor/bin/phpunit
+    `npm test`
+
+Make sure your code is linted with phpcs (PSR-2 Coding Style)
+
+    `npm run lint`
+
+You can automatically fix lint errors with phpcbf
+
+    `npm run lint:fix`
```
Comment (dplewis) [inline CONTRIBUTING.md]: Yeah auto linting is my friend on this one

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -63,7 +63,15 @@ Alternately you can configure a compatible test server as follows:
 
 You should now be able to execute the tests, from project root folder:
 
-    ./vendor/bin/phpunit
+    `npm test`
```
Comment (dplewis) [inline CONTRIBUTING.md]: nice catch

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -63,7 +63,15 @@ Alternately you can configure a compatible test server as follows:
...
+
+Make sure your code is linted with phpcs (PSR-2 Coding Style)
+
+    `npm run lint`
```
Comment (dplewis) [inline CONTRIBUTING.md]: nice catch

**Comment by dplewis** [comment]
@montymxb thanks for the feedback. The changes have been made

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -63,15 +63,19 @@ Alternately you can configure a compatible test server as follows:
...
 You should now be able to execute the tests, from project root folder:
 
-    `npm test`
+    ./vendor/bin/phpunit
+
+You may also run tests directly using phpunit as follows:
+
+    npm test
 
 Make sure your code is linted with phpcs (PSR-2 Coding Style)
```
Comment (montymxb) [inline CONTRIBUTING.md]: I figure just so people aren't wondering about why we would enforce PSR-2 we should make this a link. You can use http://www.php-fig.org/psr/psr-2/, which is the style guide published by the framework interop group. 

Making it...
Make sure your code is linted with phpcs ([PSR-2 Coding Style](http://www.php-fig.org/psr/psr-2/))

Comment (montymxb) [inline CONTRIBUTING.md]: And if we can add a colon after this statement, same as the line above it:
>You may also run tests directly using phpunit as follows:

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -63,15 +63,19 @@ Alternately you can configure a compatible test server as follows:
...
 
-    `npm run lint`
+    npm run lint
 
 You can automatically fix lint errors with phpcbf
```
Comment (montymxb) [inline CONTRIBUTING.md]: And if we can add a colon here at the end of this statement too that would be fantastic.

**Comment by montymxb** [comment]
Cool, thanks! Just waiting on one of the CI jobs to rerun real quick.

**Pull request merged**